controller/src/main/java/io/pravega/controller/eventProcessor/impl/EventProcessorGroupImpl.java

```diff
@@ -265,10 +264,18 @@ public void notifyProcessFailure(String process) throws CheckpointStoreException
             }
 
             for (Map.Entry<String, Position> entry : map.entrySet()) {
-                //Notify reader group about failed readers
+                // 1. Remove failed readers from Reader Group
                 log.info("{} Notifying readerOffline reader={}, position={}", this.objectId, entry.getKey(), entry.getValue());
                 readerGroup.readerOffline(entry.getKey(), entry.getValue());
+
+                // 2. Clean up reader from checkpoint store
+                log.info("{} removing reader={} from checkpoint store", this.objectId, entry.getKey());
+                checkpointStore.removeReader(process, readerGroup.getGroupName(), entry.getKey());
```

Comment:
Should'nt we follow the same pattern for `io.pravega.controller.eventProcessor.impl.EventProcessorGroupImpl#shutDown` ?

Reply:
Good Catch. Fixed.